test/generate-vdm.spec.ts

```diff
@@ -0,0 +1,173 @@
+/*!
+ * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+const warn = jest.fn(message => console.log('MOCKED WARNING: ', message));
+jest.mock('@oclif/command', () => {
+  const command = jest.requireActual('@oclif/command');
+  command.Command.prototype.warn = warn;
+  return command;
+});
+
+import { GeneratorOptions as GeneratorOptionsSDK, generatorOptionsCli as generatorOptionsSDK } from '@sap/cloud-sdk-generator';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import GenerateVdm, { AllOptions, BoolArgKeys, FlagsParsed, StringArgKeys } from '../src/commands/generate-vdm';
+import * as generateVdmUtil from '../src/utils/generate-vdm-util';
+
+const spyToGeneratorSDK = jest.spyOn(generateVdmUtil, 'toGeneratorSDK');
+
+describe('generate-vdm', () => {
+  const pathForTests = path.resolve(__dirname, __filename.replace(/\./g, '-')).replace('-ts', '');
+
+  beforeAll(() => {
+    const pathForResources = path.resolve(__dirname, 'resources', 'template-generator-vdm');
+    fs.copySync(pathForResources, pathForTests);
+  });
+
+  afterAll(() => {
+    fs.removeSync(pathForTests);
+  });
+
+  it('should generate a vdm.', async () => {
+    await GenerateVdm.run(['-i', 'edmxSource', '-o', 'generatedVdm', '--forceOverwrite', '--projectDir', pathForTests]);
+    const files = fs.readdirSync(path.join(pathForTests, 'generatedVdm', 'yy-1-socialnetworkaccount-cds-service'));
+    expect(files.length).toBeGreaterThan(0);
+  }, 60000);
+
+  it('should not miss or add keys by the split in string and boolean arguments.', function() {
+    // These keys should be the same as AllKeys. We use this to check that no key got lost in the splitting of bool and string
+    type AllKeysTest = BoolArgKeys | StringArgKeys;
+    type AllKeysInitial =keyof AllOptions;
+
+
+    let key1:AllKeysTest='forceOverwrite';
+    let key2:AllKeysInitial='projectDir';
```

Comment:
Don't forget to run the formatting command.

Reply:
Also I don't understand what this test does really.

Reply:
test removed